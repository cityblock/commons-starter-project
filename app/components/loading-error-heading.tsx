import * as React from 'react';
import { ErrorMessage } from './error-message';
import { LoadingMessage } from './loading-message';

interface IProps {
  error?: string;
  isLoading: boolean;
}

export const LoadingErrorHeading: React.StatelessComponent<IProps> = props => {
  const { error, isLoading } = props;
  if (error) {
    return <ErrorMessage text={error} />;
  } else if (isLoading) {
    return <LoadingMessage />;
  }
  return <div />;
};
