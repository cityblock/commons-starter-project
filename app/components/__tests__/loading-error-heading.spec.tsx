import * as React from 'react';
import { create } from 'react-test-renderer';
import { LoadingErrorHeading } from '../loading-error-heading';

it('renders note input correctly', () => {
  const tree = create(
    <LoadingErrorHeading
      isLoading={false}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
