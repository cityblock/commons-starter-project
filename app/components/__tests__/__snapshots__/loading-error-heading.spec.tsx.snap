// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders note input correctly 1`] = `<div />`;
